### PR TITLE
Keep alive & chat constructor update & a WIP disconnect function

### PR DESCRIPTION
**Pull request by STMiki**
I added a `const char msg[]` constructor to che `chat::Message`, and I needed a disconnect function to disconnect people who does not respond to the keep alive.

### CODE REVIEW

**Comment by STMiki** [comment]
Oh, I also added a `TickClock` which will run a function every x tick.

File: cubic-server/World.cpp [hunk 1]
```diff
@@ -60,3 +63,26 @@ void World::setLevelData(const world_storage::LevelData &value)
...
+void World::processKeepAlive()
+{
+    long id = std::chrono::system_clock::now().time_since_epoch().count();
+    forEachEntityIf(
+        [this, id](Entity *entity) {
+            Player *player = dynamic_cast<Player *>(entity);
+            if (player->keepAliveId() != 0) {
+                player->setKeepAliveIgnored(player->keepAliveIgnored() + 1);
+                if (this->_keepAliveClock.tickRate() * player->keepAliveIgnored() >= 600)
+                    player->disconnect();
```
Comment (huntears) [inline cubic-server/World.cpp:76]: Add a reason here for the disconnect

**Review by Trompettesib — DISMISSED** [review]

**Review by huntears — COMMENTED** [review]
Looks good, let's just have the disconnect from #85 here and it is ready to be merged

**Review by Trompettesib — DISMISSED** [review]

**Comment by STMiki** [comment]
I got disconnected because of a wrongly encoded packet. Maybe this one ????
image

**Comment by Trompettesib** [comment]
It's because of the sound packet

**Review by huntears — APPROVED** [review]

**Review by Trompettesib — APPROVED** [review]
Keep alive is ok but the chat crash when sending packets

**Pull request merged**